modules/core/04-channel/keeper/multihop_handshake_test.go

```diff
@@ -169,41 +170,122 @@ func (suite *MultihopTestSuite) TestChanOpenTryMultihop() {
 			)
 			portCap = suite.Z().Chain.GetPortCapability(suite.Z().ChannelConfig.PortID)
 		}, true},
-		// {"connection doesn't exist", func() {
-		// 	ibctesting.ChanOpenInit(paths[0].EndpointA, connectionHopsAZ)
-		// 	paths[1].EndpointB.ConnectionID = "notfound"
-		// 	chainZ := paths[len(paths)-1].EndpointB.Chain
-		// 	// pass capability check
-		// 	chainZ.CreatePortCapability(suite.chainB.GetSimApp().ScopedIBCMockKeeper, ibctesting.MockPort)
-		// 	portCap = chainZ.GetPortCapability(ibctesting.MockPort)
-		// }, true},
-		// {"connection is not OPEN", func() {
-		// 	ibctesting.ChanOpenInit(paths[0].EndpointA, connectionHopsAZ)
-		// 	// pass capability check
-		// 	chainZ := paths[len(paths)-1].EndpointB.Chain
-		// 	chainZ.CreatePortCapability(suite.chainB.GetSimApp().ScopedIBCMockKeeper, ibctesting.MockPort)
-		// 	portCap = chainZ.GetPortCapability(ibctesting.MockPort)
-
-		// 	//err := paths[2].EndpointB.ConnOpenInit()
-		// 	//suite.Require().NoError(err)
-		// }, false},
+		{"connection doesn't exist", func() {
+			suite.chanPath.EndpointA.ConnectionID = ibctesting.FirstConnectionID
+			suite.chanPath.EndpointZ.ConnectionID = ibctesting.FirstConnectionID
+
+			// pass capability check
+			suite.Z().Chain.CreatePortCapability(suite.Z().Chain.GetSimApp().ScopedIBCMockKeeper, ibctesting.MockPort)
+			portCap = suite.Z().Chain.GetPortCapability(ibctesting.MockPort)
+		}, false},
+		{"connection is not OPEN", func() {
+			suite.coord.SetupClients(suite.chanPath)
+			// pass capability check
+			suite.Z().Chain.CreatePortCapability(suite.Z().Chain.GetSimApp().ScopedIBCMockKeeper, ibctesting.MockPort)
+			portCap = suite.Z().Chain.GetPortCapability(ibctesting.MockPort)
+
+			err := suite.chanPath.EndpointZ.ConnOpenInit()
+			suite.Require().NoError(err)
+		}, false},
+		{"consensus state not found", func() {
+			suite.SetupConnections()
+			suite.chanPath.SetChannelOrdered()
+			err := suite.A().ChanOpenInit()
+			suite.Require().NoError(err)
+
+			suite.Z().Chain.CreatePortCapability(suite.Z().Chain.GetSimApp().ScopedIBCMockKeeper, ibctesting.MockPort)
+			portCap = suite.Z().Chain.GetPortCapability(ibctesting.MockPort)
+
+			heightDiff = 3 // consensus state doesn't exist at this height
+		}, false},
+		{"channel verification failed", func() {
+			// not creating a channel on chainA will result in an invalid proof of existence
+			suite.SetupConnections()
+			portCap = suite.Z().Chain.GetPortCapability(ibctesting.MockPort)
+		}, false},
+		{"port capability not found", func() {
+			suite.SetupConnections()
+			suite.chanPath.SetChannelOrdered()
+			err := suite.A().ChanOpenInit()
+			suite.Require().NoError(err)
+
+			portCap = capabilitytypes.NewCapability(3)
+		}, false},
+		{"connection version not negotiated", func() {
+			suite.SetupConnections()
+			suite.chanPath.SetChannelOrdered()
+			err := suite.A().ChanOpenInit()
```

Comment:
You can probably combine the error check like: `suite.Require().NoError(suite.A().ChanOpenInit())` to simplify a bit though this is just a nit.

Reply:
Fixed